### PR TITLE
feat(components): [cascader] add max-collpase-tags prop 

### PR DESCRIPTION
**Pull request by chenweiyi**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

## Description

Show more than one tag  when multiple and collapse-tags prop use.

### <samp>🤖 Generated by Copilot at c58d594</samp>

Added a new `max-collapse-tags` attribute to the cascader component, allowing users to limit the number of tags displayed in multiple selection mode. Updated the documentation, the component code and the example to reflect this feature.

## Explanation of Changes

+ add `max-collapse-tags` prop
+ when set `max-collapse-tags`, `presentTags` will change to `max-collapse-tags`'s value. `collapse-tag` number will change to total tags subtract max-collapse-tags。
+ update en-US doc

### <samp>🤖 Generated by Copilot at c58d594</samp>

*  Add `maxCollapseTags` prop to cascader component to control the number of tags displayed when `collapse-tags` is true ([link](https://github.com/element-plus/element-plus/pull/13386/files?diff=unified&w=0#diff-57c0ce27583f8ab4b76b01d40a154b1df1c377c1833f473446a67835328b3f2fR63-R69), [link](https://github.com/element-plus/element-plus/pull/13386/files?diff=unified&w=0#diff-eb5f57b77c2cd8863d3ab5c75d01d61d76747ab9b175c69901d4055a9889c14fL99-R101), [link](https://github.com/element-plus/element-plus/pull/13386/files?diff=unified&w=0#diff-eb5f57b77c2cd8863d3ab5c75d01d61d76747ab9b175c69901d4055a9889c14fL425-R438))
* Update documentation for cascader component to include the new prop and its usage ([link](https://github.com/element-plus/element-plus/pull/13386/files?diff=unified&w=0#diff-f66eff2fc058025d7436f3bf405e5b09e159d9c463cb2c6d47963d4877bfc0b1L89-R89), [link](https://github.com/element-plus/element-plus/pull/13386/files?diff=unified&w=0#diff-f66eff2fc058025d7436f3bf405e5b09e159d9c463cb2c6d47963d4877bfc0b1R171))
* Add example for cascader component with `maxCollapseTags` set to 3 in `docs/examples/cascader/multiple-selection.vue` ([link](https://github.com/element-plus/element-plus/pull/13386/files?diff=unified&w=0#diff-94a923bec25012f21960103513d3ed1bb95c1d32dadcfdeb298dfe1d5c24ab5bR20-R30))

### CODE REVIEW

File: packages/components/cascader/src/cascader.vue [hunk 1]
```diff
@@ -422,11 +424,18 @@ const calculatePresentTags = () => {
...
-    const [first, ...rest] = nodes
+    Array.from({ length: props.maxCollapseTags })
+      .fill(0)
+      .forEach((item, index) => {
+        const node = nodes[index]
+        if (node) {
+          tags.push(genTag(node))
+        }
+      })
+
```
Comment (btea) [inline packages/components/cascader/src/cascader.vue:435]:
```suggestion
nodes.slice(0, props.maxCollapseTags).forEach(node => {
  tags.push(genTag(node))
})
```

**Review by btea — APPROVED** [review]
LGTM

**Comment by chenweiyi** [comment]
> LGTM

happy, thank you!

**Review by ryuhangyeong — APPROVED** [review]

**Review by tolking — APPROVED** [review]

**Comment by tolking** [comment]
Thank for your contribution

**Pull request merged**